### ISSUE TITLE
Lint possible octal escape sequence in string

### ISSUE THREAD

**Issue by hellow554** [rust-lang/rust-clippy#7981]
### What it does

There is [this question](https://stackoverflow.com/questions/69981449) on Stackoverflow which has the following code:

    println!("\033[93mError\033[0m");

and the person (coming with a python background) is confused why Rust doesn't print it colored.
The obvious solution would be to use a crate to do the the work ;)

### Categories (optional)

- Kind: `clippy::suspicious`

*What is the advantage of the recommended code over the original code*

Correctly write ASCII escape characters and possible other bytes.

### Drawbacks

If you intentionnaly would like to write `"\033"` this lint would fire. But I doubt that somebody would like to insert a null-byte into their string, so this is probably worth a `#[allow(clippy::zzz)]`

### Example

```rust
println!("\033[93mError\033[0m");
```

Could be written as:

```rust
println!("\x1b[93mError\x1b[0m");
```

**Comment by birkenfeld** [comment]
I think this is a very good idea.

> I doubt that somebody would like to insert a null-byte into their string

Maybe not for strings, but for bytes literals (which should also be covered by this new lint) null bytes are quite common.

However this is easy to fix: `\x00` can be used instead in these cases, which is unambiguous.
